model/context.go

```diff
@@ -0,0 +1,40 @@
+package model
+
+import (
+	"github.com/elastic/apm-server/utility"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type Context struct {
+	service common.MapStr
+	process common.MapStr
+	system  common.MapStr
+	user    common.MapStr
+}
+
+func NewContext(service *Service, process *Process, system *System, user common.MapStr) *Context {
+	return &Context{
+		service: service.Transform(),
+		process: process.Transform(),
+		system:  system.Transform(),
+		user:    user,
+	}
+}
+
+func (c *Context) Transform(m common.MapStr) common.MapStr {
+	if m == nil {
+		m = common.MapStr{}
+	} else {
+		for k, v := range m {
```

Comment:
this reads weird at first sight, can you add a short comment saying that you do this to remove nulls? same for spanContext